src/app/data-client.ts

```diff
@@ -53,11 +53,21 @@ export class DataClient {
         break;
       }
       dataArray.push(
-        ...dataList.map((data) => ({
-          ...data.toObject(),
-          timeRequested: data.getTimeRequested()?.toDate(),
-          timeReceived: data.getTimeReceived()?.toDate(),
-        }))
+        ...dataList.map((data) => {
+          const mdIndex = data.getMetadataIndex();
+          const mdListLength = response.getMetadataList().length;
+          if (mdListLength != 0 && mdIndex >= mdListLength) {
+            throw Error(
+              `metadata index ${mdIndex} is out of response's metadata list`
+            );
+          }
+          return {
+            ...data.getData()?.toJavaScript(),
```

Comment:
can we have `data: data.getData()?.toJavaScript()`? otherwise the data is just exploded into this object and not sure if that's what we want